### DISCUSSION TITLE
Defining fill color when converting transparent PNGs

### DISCUSSION THREAD

**Discussion by gglnx** [craftcms/cms#12833] [Ideas]
Currently, converting PNG to a non-transparent format will result in a file with a black background. With #8848 was a Letterbox mode and fill configuration introduced.

This could be reused in others transform mode too to define the color which will be used when a transparent format is transformed.

**Comment by foxtrotcz** [comment]
Could you please make this change?
I have all my images transparent as webp so its as flexible as possible.
But sometimes I just need to use jpg and in such case I need to be able to select background color for the conversion.
I would like to use it in mode "fit" because i dont need exact dimensions but just fit the image but also convert the transparent color.
Currently maybe I could do it in two steps? 1. Fit without format change (keep transparent). 2. Letterbox convert to jpg and select bg color.
But this only raises the question why its not possible in single step.
Thanks.
